crates/net/src/http/query.rs

```diff
@@ -0,0 +1,34 @@
+use std::fmt;
+use wasm_bindgen::UnwrapThrowExt;
+
+/// A sequence of URL query parameters, wrapping [`web_sys::UrlSearchParams`].
+pub struct QueryParams {
+    raw: web_sys::UrlSearchParams,
+}
+
+impl Default for QueryParams {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl QueryParams {
```

Comment:
Please add the following:
- `fn from_raw(raw: web_sys::UrlSearchParams) -> Self`
- `delete` method
- `get`method
- An iterator implementation. You can move `UncheckedIter`  from headers module to gloo-utils and also use it here.

Reply:
I can add the first three methods. `UrlSearchParams` has no direct interface for iterating over the parameters, which seems to be similar to `web_sys::Headers`. Why/how does the iterator implementation for `Headers` ([here](https://docs.rs/gloo-net/latest/src/gloo_net/http/headers.rs.html#70)) work?

Reply:
There is a JS iterator implementation that `Headers` uses

Reply:
#217 can help with the 3rd point. Headers exposes `js_sys::Iterator` which we can consume from Rust

Reply:
I have added the missing methods and the iterator implementation. I temporarily moved `UncheckedIter` to the `http` module, after #217 is merged I will rebase.

Reply:
#217 has been merged. You can rebase now